stereo_points/stereo_points.py

```diff
@@ -566,112 +533,101 @@ def fiducial2Table(self, tableNode, fiducialNode):
 
-        XYZ2RAStransformation = self.transformNode_to_numpy4x4(
-            slicer.mrmlScene.GetNodesByName('leksell2RAS').GetItemAsObject(0))
+        if slicer.mrmlScene.GetNodesByName("leksell2RAS").GetNumberOfItems() == 0:
+            slicer.util.loadTransform(os.path.join(os.path.split(__file__)[0], "Resources", "leksell2RAS.h5"))
+
+        XYZ2RAStransformation = self.transformNode_to_numpy4x4(slicer.mrmlScene.GetNodesByName("leksell2RAS").GetItemAsObject(0))
         RAS2XYZtransformation = np.linalg.inv(XYZ2RAStransformation)
         return RAS2XYZtransformation
 
     def GetRAStoRASpatTrans(self):
         import numpy as np
+
         return self.transformNode_to_numpy4x4(self.frameTransform_selectionCombo.currentNode())
 
     def RAStoRASpat(self, xyz):
         import numpy as np
+
         res = np.dot(self.GetRAStoRASpatTrans(), np.array(xyz + [1])).tolist()[:3]
-        print('point in RASpat: %s' % str(res))
+        # print('point in RASpat: %s' % str(res))
         return res
 
     def GetRASpatToIJKtrans(self):
         import sitkUtils as siu
         import numpy as np
-        vol = siu.PullVolumeFromSlicer(self.referenceImage_selectionCombo.currentNode().GetName())
-        IJKtoPatRAS = np.zeros([4, 4])
-        IJKtoPatRAS[:3, :3] = np.array(vol.GetDirection()).reshape([3, 3])
-        IJKtoPatRAS[:3, 3] = np.array(vol.GetOrigin())
-        IJKtoPatRAS[3, 3] = 1
-
-        # vtkMat = vtk.vtkMatrix4x4()
-        # self.referenceImage_selectionCombo.currentNode().GetIJKToRASDirectionMatrix(vtkMat)
-        # IJKtoPatRAS = np.array([vtkMat.GetElement(i,j) for i in range(4)for j in range(4)]).reshape([4,4])
-        print("IJK2RAS: %s" % str(IJKtoPatRAS))
-
-        ##################################################################################################################
-        ########## WARNING We add the LPS2RAS matrix in between, this has been determined experimentally with the code:
-
-        ########CT_mrml = slicer.mrmlScene.GetNodesByName('CTpreop_(CT)').GetItemAsObject(0)
-        ########mat=vtk.vtkMatrix4x4()
-        ########CT_mrml.GetIJKToRASDirectionMatrix(mat)
-        ########CT_mrml = np.array([mat.GetElement(i,j) for i in range(4)for j in range(4)]).reshape([4,4])
-
-        ########CT_itk = siu.PullVolumeFromSlicer('CTpreop_(CT)')
-        ########CT_ijk2ras = np.zeros([4,4])
-        ########CT_ijk2ras[:3,:3] = np.array(CT_itk.GetDirection()).reshape([3,3])
-        ########CT_ijk2ras[:3,3] = np.array(CT_itk.GetOrigin())
-        ########CT_ijk2ras[3,3]=1
-
-        #########divide the two matrices and round:
-        ########np.dot(CT_ijk2ras, np.linalg.inv(CT_mrml))
-        ##################################################################################################################
 
+        vol = siu.PullVolumeFromSlicer(self.referenceImage_selectionCombo.currentNode().GetName())
```

Comment:
Involving SimpleITK to get volume node geometry (origin, spacing, axis directions, and extent) is both unnecessarily complicated (no need to get voxels) and insufficient (because parent transform of the volume is ignored).

You can find code snippet in the script repository that can properly transform between IJK and RAS coordinates (LPS is just inverting the sign of the first two coordinates):

https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#get-volume-voxel-coordinates-from-markup-control-point-ras-coordinates

If you want to keep things simpler and do not want to allow positioning the volume using a transform (or you only allow transforming using a linear transform but not a warping transform) then you can check the parent transform and raise an exception.